server/channels/app/migrations.go

```diff
@@ -571,6 +572,34 @@ func (s *Server) doElasticsearchFixChannelIndex() {
 	}
 }
 
+func (s *Server) doCloudS3PathMigrations() {
+	// This migration is only applicable for cloud environments
+	if os.Getenv("MM_CLOUD_FILESTORE_BIFROST") == "" {
+		return
+	}
+	// If the migration is already marked as completed, don't do it again.
+	if _, err := s.Store().System().GetByName(model.MigrationKeyS3Path); err == nil {
+		return
+	}
+
+	// If there is a job already pending, no need to schedule again.
+	// This is possible if the pod was rolled over.
```

Comment:
I guess we are not considering the cancellation case. I think it wouldn't be the case for this job because its a mandatory migration right?

Reply:
Yep, it can't be cancelled externally. But the job fully supports resumption and can handle sudden shutdowns and pod rollovers.